tests/test_core/test_alignment.py

```diff
@@ -2113,6 +2115,30 @@ def test_to_pretty(self):
         ]
         self.assertEqual(got, "\n".join(expect))
 
+    def test_to_pretty_deprecation_warning(self):
+        """produce correct pretty print formatted text"""
+        seqs = {"seq1": "ACGAANGA", "seq2": "-CGAACGA", "seq3": "ATGAACGA"}
+        expect = ["seq1    ACGAANGA", "seq2    -....C..", "seq3    .T...C.."]
+
+        aln = self.Class(data=seqs, moltype=DNA)
+        got = aln.to_pretty(name_order=["seq1", "seq2", "seq3"])
+        self.assertEqual(got, "\n".join(expect))
+
+        got = aln.to_pretty(
+            name_order=["seq1", "seq2", "seq3"], interleave_len=4
+        )  # should raise warnings here
+        expect = [
```

Comment:
this is interesting. I have tried the `self.assertRaises(DeprecationWarning)` before. I think, since that's all we want to test, the following will be cleaner

```python
with self.assertRaises(DeprecationWarning):
    aln.to_pretty(
            name_order=["seq1", "seq2", "seq3"], interleave_len=4
        )
```

will be sufficient for this test

Reply:
for some reason with `self.assertRaises(DeprecationWarning):` fails to raise errors? I've simplied the code but i'm not entirely sure why